### ISSUE TITLE
v2.5.3: no database unlock with MinimizeOnStartup=true

### ISSUE THREAD

**Issue by hmueller01** [keepassxreboot/keepassxc#4266]
Since KeePassXC 2.5.3 it is not possible to unlock the database if configuration `MinimizeOnStartup=true` is set in the GUI section. After starting KP immediately minimizes itself and can not be restored. KP 2.5.2 minimized after the password was entered and the database was unlocked. Seen this on OSX Catalina 10.15.2

## Debug Info
KeePassXC - Version 2.5.3
Revision: f8c962b

Qt 5.14.0
Diagnosemodus ist deaktiviert.

Betriebssystem: macOS 10.15
CPU-Architektur: x86_64
Kernel: darwin 19.2.0

Aktivierte Erweiterungen:
- Auto-Type
- Browser-Integration
- SSH-Agent
- KeeShare (bestätigtes und unbestätigtes Teilen)
- YubiKey
- TouchID

Kryptographische Bibliotheken:
 libgcrypt 1.8.5

**Comment by droidmonkey** [comment]
There were no changes between 2.5.2 and 2.5.3 regarding this setting. Perhaps a Qt issue.

**Comment by droidmonkey** [comment]
Duplicate of #2280

**Comment by hmueller01** [comment]
Sorry for opening a duplicate issue. I thought it was because of the new version. After installing older versions and seeing the same issue it was clear you are right. I was checking what I had installed in the mean time and could not find a sign. Than I remembered that I changes the behavior of the dock during this time (without relaunching KeePassXC). I have 2 desktops. I usually open KeepassXC in the first desktop and have two windows of the browser open (one in the first desktop, one in the second desktop). If I use autotype in the second desktop, KeePassXC switches to the first desktop, asks for the autofill entry and pastes this in the window of the browser in the **first** desktop instead of the second desktop. So I changed the behavior of KeePassXC dock to show KeePassXC on all desktops (right click on the dock icon, options, show on all desktops) which solved that problem (KeePassXC does not switch desktops when using autotype and pastes in the correct browser windows).

If I switch back dock behavior to "This desktop" or "without" (sorry, actual names can be slightly different as I have a German OSX and translated them by myself) startup with `MinimizeOnStartup=true` works like before (not minimized after startup to unlock database, but does not minimize window after unlock as I would expect).

I will post something shorter in #2280, it's just a long explanation for @droidmonkey to understand what happened.
